persistence-api/src/main/java/io/stargate/db/DbActivator.java

```diff
@@ -3,28 +3,71 @@
 import io.stargate.core.activator.BaseActivator;
 import io.stargate.db.datastore.DataStoreFactory;
 import io.stargate.db.datastore.PersistenceDataStoreFactory;
-import java.util.Collections;
+import io.stargate.db.limiter.RateLimitingManager;
+import java.util.Arrays;
+import java.util.Hashtable;
 import java.util.List;
 
+/**
+ * Activator for the {@link DataStoreFactory} service and, if enabled, the {@link
+ * RateLimitingPersistence} one.
+ *
+ * <p>For rate limiting to be activated, a service implementing {@link RateLimitingManager} first
+ * needs to be activated/registered with an "Identifier" property set to some value X, and the
+ * {@link #RATE_LIMITING_ID_PROPERTY} system property needs to be set to that X value. This is done
+ * to avoid having rate limiting activated by mistake, just because a bundle that activates a {@link
+ * RateLimitingManager} is present on the classpath (meaning, setting the {@link
+ * #RATE_LIMITING_ID_PROPERTY} acts as a confirmation that this rate limiting needs to indeed be
+ * activated).
+ */
 public class DbActivator extends BaseActivator {
-  private final ServicePointer<Persistence> persistence =
+
+  public static final String PERSISTENCE_IDENTIFIER = "StargatePersistence";
+
+  public static final String RATE_LIMITING_ID_PROPERTY = "stargate.limiter.id";
+
+  private static final String DB_PERSISTENCE_IDENTIFIER =
+      System.getProperty("stargate.persistence_id", "CassandraPersistence");
+  private static final String RATE_LIMITING_IDENTIFIER =
+      System.getProperty(RATE_LIMITING_ID_PROPERTY, "<none>");
+
+  private final ServicePointer<Persistence> dbPersistence =
       BaseActivator.ServicePointer.create(
-          Persistence.class,
-          "Identifier",
-          System.getProperty("stargate.persistence_id", "CassandraPersistence"));
+          Persistence.class, "Identifier", DB_PERSISTENCE_IDENTIFIER);
+
+  private final ServicePointer<RateLimitingManager> rateLimitingManager =
+      ServicePointer.create(RateLimitingManager.class, "Identifier", RATE_LIMITING_IDENTIFIER);
 
   public DbActivator() {
     super("DB services");
   }
 
   @Override
-  protected ServiceAndProperties createService() {
-    return new ServiceAndProperties(
-        new PersistenceDataStoreFactory(persistence.get()), DataStoreFactory.class);
+  protected List<ServiceAndProperties> createServices() {
+    Persistence persistence = this.dbPersistence.get();
+    if (!RATE_LIMITING_IDENTIFIER.equalsIgnoreCase("<none>")) {
```

Comment:
I'm not sure how well this is going to work in OSGi. If a specific `RATE_LIMITING_IDENTIFIER` is not configured, the dependency will never be satisfied and `createService` will not be invoked.

It might be preferable not to add the dependency if we know it is not configured. For example: https://github.com/stargate/stargate/pull/554/files#diff-d00f29611c415af2e85b87fe3e7559fdc9125b9d0957be87f750fce77d7c8908R180-R187